src/jit/namedintrinsiclist.h

```diff
@@ -49,7 +49,8 @@ enum HWIntrinsicFlag : unsigned int
     HW_Flag_Generic = 0x4,
     // Two-type Generic
     // - the intrinsic has two type parameters
-    HW_Flag_TwoTypeGeneric = 0xC,
+    // - Note: HW_Flag_Generic must also be set when an intrinsic has HW_Flag_TwoTypeGeneric
+    HW_Flag_TwoTypeGeneric = 0x8,
```

Comment:
I'm still not sure I understand the scenarios where we require both. It seems like it would be clearer to have those checks be: `(flag & (OneTypeGeneric | TwoTypeGeneric)) != 0`.

Reply:
Do you mean the name is confusing? 
I just thought that `TwoTypeGeneric` implying `Generic` and requiring additional generic checking, so that most of the generic intrinsic can just have `HW_Flag_Generic`. And the framework can be:
```
if ((flag & HW_Flag_Generic) != 0)
{
    // checking the type-argument for most of the generic intrinsic 
    // and the first type-argument of StaticCast<T, U>

    if ((flag & HW_Flag_TwoTypeGeneric) != 0)
    {
        // // checking the second type-argument of StaticCast<T, U>
    }
}
```

Reply:
> Do you mean the name is confusing?

No, that the dependence is confusing.

The code could be:
```
if ((flag & (HW_Flag_Generic | HW_Flag_TwoTypeGeneric)) != 0)
{
    // checking the type-argument for most of the generic intrinsic 
    // and the first type-argument of StaticCast<T, U>

    if ((flag & HW_Flag_TwoTypeGeneric) != 0)
    {
        // checking the second type-argument of StaticCast<T, U>
    }
}
```

Which is perhaps more clear, doesn't require users to remember to also set `HW_Flag_Generic` (which could then be named `HW_Flag_OneTypeGeneric`), and still only requires a single check (since `HW_Flag_Generic | HW_Flag_TwoTypeGeneric` will be folded to a constant by the compiler).

Reply:
Got it. Thanks for explaining.